File: VB/SelectPaperSource/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraPrinting
Imports System.Drawing.Printing
Imports DevExpress.XtraReports.UI
' ...

Namespace SelectPaperSource
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
	Dim pt As New ReportPrintTool(New XtraReport1())
	AddHandler pt.PrintingSystem.StartPrint, AddressOf printingSystem_StartPrint
	pt.PrintDialog()
End Sub

		Private Sub printingSystem_StartPrint(ByVal sender As Object, ByVal e As PrintDocumentEventArgs)
			For i As Integer = 0 To e.PrintDocument.PrinterSettings.PaperSources.Count - 1
				If e.PrintDocument.PrinterSettings.PaperSources(i).Kind = PaperSourceKind.TractorFeed Then
					e.PrintDocument.DefaultPageSettings.PaperSource = e.PrintDocument.PrinterSettings.PaperSources(i)
					Exit For
				End If
			Next i

			For i As Integer = 0 To e.PrintDocument.PrinterSettings.PrinterResolutions.Count - 1
				If e.PrintDocument.PrinterSettings.PrinterResolutions(i).Kind = PrinterResolutionKind.High Then
					e.PrintDocument.DefaultPageSettings.PrinterResolution = e.PrintDocument.PrinterSettings.PrinterResolutions(i)
					Exit For
				End If
			Next i
		End Sub

	End Class
End Namespace

File: CS/SelectPaperSource/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using System.Drawing.Printing;
// ...

namespace SelectPaperSource {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 rep = new XtraReport1();
            rep.PrintingSystem.StartPrint += 
                new PrintDocumentEventHandler(printingSystem_StartPrint);
            rep.PrintDialog();
        }

        private void printingSystem_StartPrint(object sender, PrintDocumentEventArgs e) {
            for (int i = 0; i < e.PrintDocument.PrinterSettings.PaperSources.Count; i++)
                if (e.PrintDocument.PrinterSettings.PaperSources[i].Kind == 
                    PaperSourceKind.TractorFeed) {
                    e.PrintDocument.DefaultPageSettings.PaperSource = 
                        e.PrintDocument.PrinterSettings.PaperSources[i];
                    break;
                }

            for (int i = 0; i < e.PrintDocument.PrinterSettings.PrinterResolutions.Count; i++)
                if (e.PrintDocument.PrinterSettings.PrinterResolutions[i].Kind == 
                    PrinterResolutionKind.High) {
                    e.PrintDocument.DefaultPageSettings.PrinterResolution = 
                        e.PrintDocument.PrinterSettings.PrinterResolutions[i];
                    break;
                }
        }
    }
}